### ISSUE TITLE
Ensure that pkg.installed function refreshes properly with windows.

### ISSUE THREAD

**Issue by hrumph** [saltstack/salt#34767]
### Description of Issue/Question
A while back pkg.latest was fixed and I believe works properly but no one got around to fixing pkg.installed. In pkg.installed, if the version was 'latest', a refresh will happen no matter what
(possibly in addition to others) because of the lines

```
if salt.utils.is_true(kwargs.get('refresh', True)):
        refresh_db(saltenv)
```

in pkg.latest_version. Also if the version is not 'latest' and the cache was not already refreshed the state would fail because the call to pkg.list_pkgs (called by _find_install_targets) requires that the pkg database exist in order to work.

**Comment by rallytime** [comment]
Thanks for filing this bug @hrumph and for the fix. Love it when that happens! :)

**Comment by hrumph** [comment]
The fix may not be fully testable in 2015.8 because fixes for use of 'latest' were made to 2016.3 (see #31114)
